src/overlays/actors/ovl_Bg_Icicle/z_bg_icicle.c

```diff
@@ -21,26 +52,199 @@ const ActorInit Bg_Icicle_InitVars = {
     (ActorFunc)BgIcicle_Update,
     (ActorFunc)BgIcicle_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Icicle_0x809C9A60/BgIcicle_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 1500, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -3, ICHAIN_CONTINUE),
+    ICHAIN_F32(minVelocityY, -30, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx D_060000D0[];
+extern CollisionHeader D_06000294;
+
+void BgIcicle_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIcicle* this = THIS;
+
+    s32 paramsHigh;
+    s32 paramsMid;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000294);
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+
+    paramsHigh = (thisx->params >> 8) & 0xFF;
+    paramsMid = (thisx->params >> 2) & 0x3F;
+    this->unk_161 = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 3;
```

Comment:
```suggestion
    thisx->params = thisx->params & 3;
```
Would be easier to see, and add a params MACRO later.

Reply:
I'm used to writing it like this universally, although indeed in the case of compound params it may be worth macroing this too.